CMakeLists.txt

```diff
@@ -1,124 +1,232 @@
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_index_cpp REQUIRED)
+find_package(mpi_cmake_modules REQUIRED)
+find_package(pybind11 REQUIRED)
+find_package(cereal REQUIRED)
+find_package(robot_interfaces REQUIRED)
+find_package(serialization_utils REQUIRED)
+find_package(yaml_utils REQUIRED)
+find_package(sensor_msgs REQUIRED)
+
+find_package(Eigen3 REQUIRED)
 find_package(OpenCV REQUIRED)
 
 find_package(Pylon)
 if (${Pylon_FOUND})
     add_definitions(-DPylon_FOUND)
-
-    # Set library names to variables, so we can use the variable instead of the
-    # direct name below.  In case, Pylon drivers are not built, the variables
-    # will be empty and thus not cause trouble when used for linking etc.
-    set(pylon_driver pylon_driver)
-    set(tricamera_driver tricamera_driver)
+    ament_export_dependencies(Pylon)
 endif()
 
 
+ament_python_install_package(${PROJECT_NAME} PACKAGE_DIR python/${PROJECT_NAME})
 
-search_for_eigen()
 
-
-catkin_python_setup()
-
-catkin_package(
-    INCLUDE_DIRS include ${Pylon_INCLUDE_DIRS}
-    LIBRARIES
-        camera_observations
-        opencv_driver
-        ${pylon_driver}
-        ${tricamera_driver}
-        pybullet_tricamera_driver
-        camera_calibration_parser
-    CFG_EXTRAS
-        cfg_extras.cmake
-)
-
-
-include_directories(
-    include
-    ${catkin_INCLUDE_DIRS}
-    ${Pylon_INCLUDE_DIRS}
-    ${OpenCV_INCLUDE_DIRS}
-)
+ament_export_interfaces(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
 
 
 # Libraries
 add_library(camera_observations
     src/camera_observation.cpp
     src/tricamera_observation.cpp
 )
-target_link_libraries(camera_observations ${catkin_LIBRARIES} ${OpenCV_LIBRARIES})
+target_include_directories(camera_observations PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+    ${OpenCV_INCLUDE_DIRS})
+target_link_libraries(camera_observations
+    ${OpenCV_LIBRARIES}
+    serialization_utils::serialization_utils
+)
+
 
 add_library(opencv_driver src/opencv_driver.cpp)
+target_include_directories(opencv_driver PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+    ${OpenCV_INCLUDE_DIRS}
+)
 target_link_libraries(opencv_driver
-    ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} camera_observations
+    ${OpenCV_LIBRARIES}
+    robot_interfaces::robot_interfaces
+    camera_observations
 )
 
+
 if (${Pylon_FOUND})
-    add_library(${pylon_driver} src/pylon_driver.cpp)
-    target_link_libraries(${pylon_driver}
-        ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} ${Pylon_LIBRARIES}
+    add_library(pylon_driver src/pylon_driver.cpp)
+    target_include_directories(pylon_driver PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+        ${OpenCV_INCLUDE_DIRS}
```

Comment:
Isn't there a modern cmake target for opencv now?
just wondering...

Reply:
From what I found no.  I didn't explicitly test it though.